lib/rgeo/geos/capi_feature_classes.rb

```diff
@@ -69,6 +69,44 @@ def as_text
         str
       end
       alias to_s as_text
+
+      # Constructs a Voronoi diagram from the vertices of the input geometry.
+      #
+      # The source may be any geometry type. All vertices of the geometry will be
+      # used as the input points to the diagram.
+      #
+      # @param enevelope [RGeo::Feature::Geometry | nil]
+      #   The +envelope+ keyword argument provides an envelope to use to clip the
+      #   resulting diagram. If +nil+, it will be calculated automatically.
+      #   The diagram will be clipped to the *larger* of the provided envelope
+      #   or an envelope surrounding the sites.
+      #
+      # @param tolerance [Float]
+      #   The +tolerance+ keyword argument sets the snapping tolerance used to improve
+      #   the robustness of the computation. A tolerance of 0.0 specifies
+      #   that no snapping will take place. The +tolerance+ argument can be
+      #   finicky and is known to cause the algorithm to fail in several cases.
+      #   If you're using +tolerance+ and getting a failure, try removing it.
+      #   The test +test_voronoi_diagram_tolerance+ in +test/common/voronoi_tests.rb+
+      #   shows more details.
+      #
+      # @param only_edges [true|false]
+      #   If the +only_edges+ keyword argument is +false+ a collection of +Polygon+s
+      #   will be returned. Otherwise a collection of +LineString+ edges is returned.
+      #
+      # @return [RGeo::Feature::Geometry]
+      #   if +only_edges+ is true, this will return a LineString or MultiLineString.
+      #   Otherwise, it will be a GeometryCollection of polygons.
```

Comment:
The fact that it has 3 different possible return values may be bothering, WDYT?

Reply:
I think that's fine. There's other methods like `union` that will return different types depending on the resultant geometry graph so this isn't unusual. Does accounting for the different types make it more difficult to handle in the method?

Reply:
Not at all. It may just make it more difficult in the usage. Especially the linestring vs multilinestring seems tricky. But to be fair, this may be a geos issue since it also affects geos and shapely!

Reply:
https://github.com/libgeos/geos/pull/702 Geos PR